pkg/finddrive/finddrive.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package finddrive provides functionality to find an NVMe block device associated with
+// a particular physical slot on the machine, based on information in the SMBIOS table.
+package finddrive
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/smbios"
+)
+
+const (
+	// M2MKeySlotType is the SMBIOS slot type code for M.2 M-key
+	M2MKeySlotType = 0x17
+)
+
+// The path /sys/block/nvme0n1 will be a symlink to
+// /sys/devices/pci<domain:bus>/<slot BDF>/<drive BDF>/nvme/nvme0/nvme0n1
```

Comment:
worth adding additional comment for an anatomy by example highlighting which are  1) domain bus str 2) slot BDF 3) drive BDF

```
devices/pci0a5a:44/0a5a:44:12.6/0a5a:00:00.0/nvme/nvme0/

domain:bus    0a5a:44
slot BDF        = {domain:bus} : {dev} . {fn} = 0a5a:44:12.6
                        {domain:bus} = 0a5a:44
                        {dev} = 12
                        {fn} = 6
                       
```

otherwise, I find it a bit hard to visualize the format code in line 29

fmt.Sprintf("devices/pci%s/%s:%02x.%x/", domainBusStr, domainBusStr, dev, fn)

if you have other ideas to make line 29, esp, this format str more readable, it'd be great

Reply:
Expanded the comment to explain this more clearly.